core/src/main/java/org/elasticsearch/action/search/SearchDfsQueryAndFetchAsyncAction.java

```diff
@@ -46,9 +47,10 @@
     SearchDfsQueryAndFetchAsyncAction(ESLogger logger, SearchTransportService searchTransportService,
                                               ClusterService clusterService, IndexNameExpressionResolver indexNameExpressionResolver,
                                               SearchPhaseController searchPhaseController, ThreadPool threadPool,
-                                              SearchRequest request, ActionListener<SearchResponse> listener) {
+                                              SearchRequest request, ActionListener<SearchResponse> listener,
+                                              PluginsService pluginsService) {
```

Comment:
Rather than pass `PluginsService` down I think it'd be more in keeping with how we've been changing the plugins to make a new interface like:
```
public interface SearchListener {
  void onFinishedSearchResponse(stuff);
}
```

Then make a method in `SearchPlugin` like
```
List<SearchListener> getSearchListeners() {return emptyList()
```

Then extract all the search listeners in `SearchModule` and throw that list around.

It is more stuff but then you don't need to pass around `PluginsService` which is really a startup-time thing.